.github/workflows/main.yml

```diff
@@ -63,3 +81,25 @@ jobs:
           git -c user.name='ci' -c user.email='ci' commit -m init
           git push -f -q https://git:${{ secrets.github_token }}@github.com/${{ github.repository }} HEAD:gh-pages
         if: github.event_name == 'push' && github.event.ref == 'refs/heads/master'
+
+  msrv:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-14, windows-latest]
+    env:
+      MSRV: 1.63
+    steps:
+    - uses: actions/checkout@master
+    - name: Install Rust (rustup)
+      run: |
+        rustup toolchain install $MSRV --no-self-update --profile minimal
+        rustup default $MSRV
+        rustup toolchain install nightly --no-self-update --profile minimal
+      shell: bash
+
+    - run: cargo +nightly update -Zminimal-versions
```

Comment:
Please instead commit your lockfile

Reply:
I'm not sure committing lockfile is better than `-Zminimal-versions`

`-Zminimal-versions` make sure the minimal version of dependency is used, committing a lockfile will then require user to maintain it whenever they update their dependencies and re-run `cargo +nightly -Zminimal-versions update` locally.

Reply:
The convention here is to check in a `Cargo.minimal-versions.lock` file or `Cargo.msrv.lock` and use that in this CI job instead

Reply:
Thanks, I've updated the PR to use `Cargo.msrv.lock`

Reply:
No, please do not do that either!  That significantly increases maintenance / degrades the contributor's experience.

For why to commit your lockfile, see https://doc.rust-lang.org/stable/cargo/faq.html#why-have-cargolock-in-version-control

Maintaining a `Cargo.msrv.lock` means you need to have a `--locked` job for that and anytime you use it (or edit `Cargo.toml`), you need to juggle it in and out of your regular `Cargo.lock`.

In the MSRV RFC, I split this up into [two extended MSRV workflows](https://rust-lang.github.io/rfcs/3537-msrv-resolver.html#extended-msrv) and found very little motivation to justify the two-lockfile solution.  Just keep your `Cargo.lock` as MSRV compatible and have a non-blocking "latest versions" verification job.

Reply:
I've renamed `Cargo.msrv.lock` back to `Cargo.lock` and commit it.

Please review again if you have time @epage